openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/image/ImageSignatureOperationsImpl.java

```diff
@@ -39,7 +39,7 @@ public ImageSignatureOperationsImpl(OperationContext context, String apiGroupNam
   @Override
   public Boolean delete() {
     try {
-      handleDelete(getResourceUrl(), context.getGracePeriodSeconds(), context.getPropagationPolicy(), context.getCascading());
+      handleDelete(getResourceUrl(), context.getGracePeriodSeconds(), context.getPropagationPolicy(), null, context.getCascading());
```

Comment:
Why null instead of `context.getResourceVersion()`?

@rohanKanojia do you know why we have a specific ImageSignatureOperationsImpl#delete override?

Reply:
My confusion here is that I assumed that ImageSignature is just a KubernetesResource because the operation extends from CreateOnlyResourceOperationsImpl, which is not a HasMetadataOperation - however it is, so the resourceVersion is applicable.

We can make the change here to pass the resourceVersion, but I'd vote for either having CreateOnlyResourceOperationsImpl be a HasMetadataOperation, or to remove CreateOnlyResourceOperationsImpl altogether.  

With #3327 as it stands you can't register anything that extends CreateOnlyResourceOperationsImpl as an operation (this is case currently as well, there's no ImageSignatureResourceHandler that is generated).  

To remove it entirely means creating an interface to replace references to NamespacedCreateOnlyResourceOperationsImpl and that if you use resource(ImageSignature) - you could attempt the other write operations, but they'd simply fail.

Reply:
OK, I missed the `CreateOnlyResourceOperationsImpl`.

There are some non-standard Kubernetes->OpenShift resources that don't have full CRUD support. I think we started to add proper OperationsImpl to those in the scope of the latest OpenShift resource type implementations (#3223).

ImageSignatures are deletable, createable, but cannot be replaced.

Regarding #3327, I don't know what the best way to handle this situation would be.

Reply:
> Regarding #3327, I don't know what the best way to handle this situation would be.

Would you be opposed to removing CreateOnlyResourceOperationsImpl?  For standard dsl calls it should be sufficient that the OperationImpl or associated (NamespacedCreateOnlyResourceOperationsImpl) is accessed through the relevant interfaces.

Reply:
Implementation-wise I think it's not important.

What we should preserve is the DSL syntax, i.e. there's no `client.imageSignatures().withName("is").replace()` method (neither `edit`, `patch`, etc.).

Having the DSL method return a proper interface sound like a cleaner solution for me, yes.

Reply:
Not sure if the original sugestion is needed anymore, but changed it anyway. Please let me know if this should be reverted.